internal/store/hybrid.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"context"
+	"io"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+)
+
+type hybrid struct {
+	oras.ReadOnlyTarget
+	combined []oras.Target
+}
+
+// NewReadOnlyHybrid generates a new hybrid storage.
+func NewReadOnlyHybrid(provider oras.ReadOnlyTarget, combined ...oras.Target) oras.ReadOnlyTarget {
```

Comment:
It would be better to have something like [`io.MultiReader`](https://pkg.go.dev/io#MultiReader).

It can be
```go
func MultiReadOnlyTarget(targets ...oras.ReadOnlyTarget) oras.ReadOnlyTarget
```

Reply:
should it be
```golang
func MultiReadOnlyTarget(provider oras.ReadOnlyTarget, targets ...oras.ReadOnlyTarget) oras.ReadOnlyTarget
```